dali/kernels/common/join/tensor_join_cpu_test.cc

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include "dali/kernels/common/join/tensor_join_cpu.h"
+
+namespace dali {
+namespace kernels {
+namespace test {
+
+TEST(TensorJoinCpuTest, DetermineShapeStack) {
+  std::vector<TensorShape<>> shin = {{4, 5, 7, 8},
+                                     {4, 5, 7, 8}};
+  TensorShape<> sh0 = {2, 4, 5, 7, 8};
+  TensorShape<> sh1 = {4, 2, 5, 7, 8};
+  TensorShape<> sh2 = {4, 5, 2, 7, 8};
+  TensorShape<> sh3 = {4, 5, 7, 2, 8};
+  TensorShape<> sh4 = {4, 5, 7, 8, 2};
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 0), sh0);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 1), sh1);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 2), sh2);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 3), sh3);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 4), sh4);
+}
+
+
+TEST(TensorJoinCpuTest, DetermineShapeConcatConstantExtent) {
+  std::vector<TensorShape<>> shin = {{4, 5, 7, 8},
+                                     {4, 5, 7, 8}};
+  TensorShape<> sh0 = {8, 5, 7, 8};
+  TensorShape<> sh1 = {4, 10, 7, 8};
+  TensorShape<> sh2 = {4, 5, 14, 8};
+  TensorShape<> sh3 = {4, 5, 7, 16};
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 0), sh0);
```

Comment:
```suggestion
  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 0), {8, 5, 7, 8});
```
and so on would read a bit easier (just a suggestion)

Reply:
I wanted to emphasise, which reference data goes for which dimension